packages/malloy/src/dialect/duckdb.ts

```diff
@@ -357,16 +354,22 @@ export class DuckDBDialect extends Dialect {
     n: Expr,
     timeframe: DateUnit
   ): Expr {
+    let repeat = 1;
     if (timeframe == "quarter") {
       timeframe = "month";
-      n = mkExpr`${n}*3`;
+      repeat = 3;
     }
     if (timeframe == "week") {
       timeframe = "day";
-      n = mkExpr`${n}*7`;
+      repeat = 7;
     }
     const interval = mkExpr`INTERVAL ${n} ${timeframe}`;
-    return mkExpr`((${expr.value})${op}${interval})`;
+    const opInterval = mkExpr`${op} ${interval}`;
+    const repeatedOpInterval = Array.from(
+      { length: opInterval.length * repeat },
+      (_, i) => opInterval[i % opInterval.length]
+    );
+    return mkExpr`((${expr.value}))${repeatedOpInterval}`;
```

Comment:
@mtoy-googly-moogly This is the part that generates `time + INTERVAL N DAY + INTERVAL N DAY + INTERVAL N DAY + INTERVAL N DAY + INTERVAL N DAY + INTERVAL N DAY + INTERVAL N DAY` to add `N` weeks... Any other ideas how to implement this?

Reply:
I don't understand why this isn't 

    const interval = mkExpr`INTERVAL ${n * repeat} ${timeframe}`

Reply:
( just edited ^^^ )

Reply:
So it can't be literally that because `n` is an `Expr`, not a `number`. But we can do 

```ts
if (timeframe == "quarter") {
  n = mkExpr`${n}*3`;
  timeframe = "month";
} else if (timeframe == "week") {
  n = mkExpr`${n}*7`;
  timeframe = "day";
}
const interval = mkExpr`INTERVAL (${n}) ${timeframe}`
```

Which is identical to what you had in the first place, except I've added parentheses around the expression when creating the interval. DuckDB didn't seem to like `INTERVAL 1*7 DAY` that we were generating, and I couldn't find any examples of using expressions in the interval creation syntax, so I thought that it had to be a literal. It doesn't, but in this case it does have to be in parentheses.